### PR TITLE
JR2-82 Rework occupation model

### PR DESCRIPTION
**Pull request by michael-wirth**
- rework occupation model
- load and index initial data

### CODE REVIEW

File: src/main/java/ch/admin/seco/service/reference/service/impl/OccupationServiceImpl.java [hunk 1]
```diff
@@ -129,22 +134,31 @@ public void delete(UUID id) {
...
+    }
 
-            )
-            .collect(Collectors.toList());
+    @PostConstruct
+    private void init() {
+        occupationServiceImpl = applicationContext.getBean(OccupationServiceImpl.class);
+    }
+
+    private OccupationSynonym createSuggestionLists(Occupation occupation) {
```
Comment (birom) [inline src/main/java/ch/admin/seco/service/reference/service/impl/OccupationServiceImpl.java]: I think the name is misleading.

File: src/main/java/ch/admin/seco/service/reference/service/impl/OccupationServiceImpl.java [hunk 2]
```diff
@@ -129,22 +134,31 @@ public void delete(UUID id) {
...
+        return new OccupationSynonym()
+            .id(occupation.getId())
+            .code(occupation.getCode())
+            .language(occupation.getLanguage())
+            .occupation(occupation.getName())
+            .occupationSuggestions(createSuggestions(occupation.getName()));
```
Comment (birom) [inline src/main/java/ch/admin/seco/service/reference/service/impl/OccupationServiceImpl.java]: Why don't you delegate the call to the ElasticsearchIndexServiceImpl? This method is dupplicated there.

Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/reference/service/impl/OccupationServiceImpl.java]: done. i extracted the functionality into a MapperHelper class

File: src/main/java/ch/admin/seco/service/reference/service/impl/OccupationServiceImpl.java (hunk 1 above)
Comment (michael-wirth) [inline src/main/java/ch/admin/seco/service/reference/service/impl/OccupationServiceImpl.java]: fixed

**Pull request merged**